src/main.rs

```
@@ -147,6 +157,24 @@ struct VscodeSearchProvider {
 }
 
 impl VscodeSearchProvider {
+    /// Add a workspace.
+    fn add_workspace(&mut self, url: String) {
+        if let Some(name) = url.split('/').last() {
```

I must admit that I don't like that this function can silently fail to add a namespace if there's no final name component (i.e. `.last()` returns `None`).  I know it's the same behaviour as before, but now that it's a function I think it deserves a bit more error handling.

Would you mind to make this function return a `Result<()>`, and return `Ok(())` if it got added and `Err(anyhow!("Failed to extract workspace name from uri {}", uri))` otherwise?

---

If I do that, I need to handle the return value. How should that be done? Via map_err, error! and zbus::fdo::Error, like some other places?

---

I've pushed some changes. Not sure if that's the preferred way to do this.